### PR TITLE
Allow setting default container image registry in configuration

### PR DESCRIPTION
**Pull request by kbknapp**
This PR allows setting the default container image registry in the
configuration file. This will be used if not provided at the CLI for commands
that require a container image reference.

The configuration is set as follows (showing the default value):

```toml
[seaplane]
default-registry = "registry.cplane.cloud/"
```

### CODE REVIEW

**Comment by kbknapp** [comment]
Current dependencies on/for this PR:
* main
  * **PR #254** <a href="https://app.graphite.dev/github/pr/seaplane-io/seaplane/254" target="_blank">Graphite</a>  👈
    * **PR #255** <a href="https://app.graphite.dev/github/pr/seaplane-io/seaplane/255" target="_blank">Graphite</a> 
      * **PR #256** <a href="https://app.graphite.dev/github/pr/seaplane-io/seaplane/256" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/seaplane-io/seaplane/254?utm_source=stack-comment).

File: docs/CONFIGURATION_SPEC.md [hunk 1]
```diff
@@ -44,23 +75,40 @@ interactions with the APIs.
...
+- `ansi`: If output is selected, always use ANSI color codes even on Windows
+  (instead of console API calls)
 - `never`: Do not color output
 
+#### The `default-registry` Field
+
+Names which container image registry will be inferred if not provided for
+commands and CLI flags that take a container image reference.
+
+Default value: `registry.hub.docker.com/`
```
Comment (RafalGoslawski) [inline docs/CONFIGURATION_SPEC.md:93]: This differs from PR description. Which one is correct? 
```
The configuration is set as follows (showing the default value):

[seaplane]
default-registry = "registry.cplane.cloud/"
```

**Review by RafalGoslawski — APPROVED** [review]
LGTM, mod mismatch between registry in PR description and code

**Comment by kbknapp** [comment]
Ah got ahead of myself in the description #255 actually changes the default. I did it in two PRs to try and make the diffs easier to track. 😅

**Review by ericsampson — APPROVED** [review]
- Could we maybe change the name to `default-registry-url` to match the others (and just be a tiny bit more descriptive)
- I'm sure the trailing slash requirement is going to bite people lol

**Comment by ericsampson** [comment]
Thanks Kevin!

**Comment by ericsampson** [comment]
>  can add code to check for it and make that requirement transparent to the user.

I know it's probably overkill, but maybe in the existing unit tests you could make it use both 'ends in slash and not' in some of the tests, just to get coverage?

**Comment by kbknapp** [comment]
> I know it's probably overkill, but maybe in the existing unit tests you could make it use both 'ends in slash and not' in some of the tests, just to get coverage?

I'll probably have to circle back to this because the conversion doesn't happen until we go from a `RawConfig` -> `Ctx` which currently doesn't have any tests. It's on my list, just time and priorities 😅

**Comment by ericsampson** [comment]
100%, that's why I said it's overkill for such a simple change, no worries lol

**Pull request merged**